enhancements/installer/gcp-ipi-shared-vpc.md

```diff
@@ -0,0 +1,262 @@
+api-int, and \*.apps DNS records.
+
+After all of these resources are created, they will be used in a manually
+created install-config.yaml to configure the cluster accordingly. At a minimum,
+the credentialsMode will be set to Passthrough, and the network,
+computeNetwork, controlPlaneNetwork, and networkProjectID will be configured to
+reference the Shared VPC network. When minimial permissions are desired in the
+Host Project, the additional parameters will be configured to enable the
+cluster to consume them as well.
+
+The installation process can then proceed as any other IPI workflow by running
+the create manifests, create ignition, and/or create cluster sub-commands.
+
+### API Extensions
+
+#### api.dns
+
+The current DNSZone struct does not appear to be able to store which project
+the zone is in. The ID variable can be used to store fully qualified path
+names, which will include the project. However, this may require modification
+of all consumers of this variable to be able to parse this information.
+Alternatively, a second variable can be added to store the projectID and the
+consumers can use that information when linking to the DNS Zone.
+
+#### api.ingress
+
+The current ingress structures do not appear to be able to store an address to
+be used when creating the ingress load balancer(s). New field(s) may be needed
+to enable this functionality. They can then be passed to the gcp lagacy
+provider for creating the ingress load balancers.
+
+#### installconfig.platform.gcp.networkProjectID
+
+A new parameter will be added to the install-config to specify the project id
+of the network and subnets. This parameter will then be used where necessary to
+validate existing network project resources as well as provision additional
+resources using these network resources. The default will be an empty string
+and will indicate these resources (should) exist in the default cluster
+project.
+
+#### installconfig.platform.gcp.privateManagedZone
+
+A new parameter will be added to the install-config to specify the private DNS
+managed zone. This parameter will be used where necessary to validate existing
+DNS resources as well as provision additional private DNS records. The default
+will be an empty string, which will indicate these resources are to be created
+and managed in the project identified by the networkProjectID. When this
+parameter is set to “none”, the cluster will not be configured to manage
+private DNS resources.
+
+#### installconfig.platform.gcp.publicManagedZone
+
+A new parameter will be added to the install-config to specify the public DNS
+managed zone. This parameter will be used where necessary to validate existing
+DNS resources as well as provision additional public DNS records. The default
+will be an empty string, which will indicate these resources are to be created
+and managed in the project identified by the networkProjectID (when publish is
```

Comment:
The public DNS zone can also go in the service project, right? 

1. Do we have a sense of whether users prefer creating the zones in the host/service project?
2. Would we create a knob to decide which project the public zone is going in?

Reply:
The public zone can go in either, or an unrelated project. The knob we are providing is going to be the selfLink to an existing zone, which will include the project name of which it belongs.